setup.py

```diff
@@ -108,8 +108,15 @@ def get_version_info():
     elif os.path.exists('scipy/version.py'):
         # must be a source distribution, use existing version file
         # load it as a separate module to not load scipy/__init__.py
-        import imp
-        version = imp.load_source('scipy.version', 'scipy/version.py')
+        # based on approach described in python-ideas
+        # https://mail.python.org/pipermail/python-ideas/2014-December/030265.html
+        import importlib
+        module_name = 'scipy.version'
+        path = 'scipy/version.py'
+        loader = importlib.machinery.SourceFileLoader(module_name, path)
+        spec = importlib.machinery.ModuleSpec(module_name, loader, origin=path)
+        module = importlib.util.module_from_spec(spec)
+        version = loader.exec_module(module)
```

Comment:
this updated block is adapted somewhat blindly from the commentary on python ideas mailing list--haven't checked how well this code path is flushed by tests in one way or another